### DISCUSSION TITLE
Conditional ProjectReference / PackageReference pattern recomendations

### DISCUSSION THREAD

**Discussion by p-m-j** [dotnet/project-system#9272] [Q&A]
When working on a project which has a dependency on an internal package which lives in another source code repository it is ocassionally useful to swap out the PackageReference for a ProjectReference to iterate faster. 

To date the best approach I have come up with to accomodate this (or hacked together from stackoverflow et al.) is as follows.

1. Maintain multiple solution files, one with only the projects from the current repository and another with additional projects from the other repository.
2. Make use of MSBuild properties and the Choose element to conditionaly add either a project or a package reference.
```xml
<Project>
  <PropertyGroup>
    <UseProjectReferences>true</UseProjectReferences>
  </PropertyGroup>
  <Choose>
    <When Condition="'$(UseProjectReferences)'=='true' And Exists('..\..\..\foo-repo-root')">
      <ItemGroup>
        <ProjectReference Include="..\..\..\foo-repo-root\src\Foo\Foo.csproj" />
      </ItemGroup>
    </When>
    <Otherwise>
      <ItemGroup>
        <PackageReference Include="Foo" Version="0.0.0-beta.*" />
      </ItemGroup>
    </Otherwise>
  </Choose>
</Project>
```
This works reasonably well, the Exists condition being the latest addition in order to reduce errors when cloned to another developers machine.

Are there further improvements that can be made? 

Is there any way to lose the additional solution file?
~~Is there anyway to set the UseProjectReferences build property based on the solution being used~~ I found $(SolutionName) whilst writing this post.

**Comment by drewnoakes** [comment]
See also https://github.com/dotnet/project-system/issues/195
